pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -164,22 +144,18 @@ def __init__(
             raise ValueError(
                 "`sampling_rate` is too small, less than 4 data points."
             )
-        self._sampling_rate = sampling_rate
-        self.sampling_times = self._adapt_to_sampling_rate(
-            # Include extra time step for final instruction from samples:
-            np.arange(self._tot_duration + 1, dtype=np.double)
-            / 1000
+        # Sets the config as well as builds the hamiltonian
+        self.hamiltonian = Hamiltonian(
```

Comment:
Are we sure about ` hamiltonian` being a public attribute? To me, it makes more sense to hide it

Reply:
I am sure of nothing haha I made it public because I tought it could be useful to access this attribute, as well as I wanted to distinguish between _hamiltonian (inner property of hamiltonian) and hamilotonian, but since I am not testing it and since `_hamiltonian` was a private property here, let's keep it private.